File: Backend/Model/Generated.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Newtonsoft.Json;

namespace VirtualVoid.Model {
  public class Types {
    
    #region BackLog
    public class BackLog {
      #region members
      [JsonProperty("id")]
      public string id { get; set; }
    
      [JsonProperty("name")]
      public string name { get; set; }
    
      [JsonProperty("issues")]
      public List<Issue> issues { get; set; }
      #endregion
    }
    #endregion
    
    #region Board
    public class Board {
      #region members
      [JsonProperty("id")]
      public string id { get; set; }
    
      [JsonProperty("backlog")]
      public BackLog backlog { get; set; }
    
      [JsonProperty("states")]
      public List<State> states { get; set; }
      #endregion
    }
    #endregion
    
    #region BoardInput
    public class BoardInput {
      #region members
      public string name { get; set; }
      #endregion
    
      #region methods
      public dynamic GetInputObject()
      {
        IDictionary<string, object> d = new System.Dynamic.ExpandoObject();
    
        var properties = GetType().GetProperties(System.Reflection.BindingFlags.Instance | System.Reflection.BindingFlags.Public);
        foreach (var propertyInfo in properties)
        {
          var value = propertyInfo.GetValue(this);
          var defaultValue = propertyInfo.PropertyType.IsValueType ? Activator.CreateInstance(propertyInfo.PropertyType) : null;
    
          var requiredProp = propertyInfo.GetCustomAttributes(typeof(JsonRequiredAttribute), false).Length > 0;
    
          if (requiredProp || value != defaultValue)
          {
            d[propertyInfo.Name] = value;
          }
        }
        return d;
      }
      #endregion
    }
    #endregion
    
    #region Epic
    public class Epic {
      #region members
      [JsonProperty("id")]
      public string id { get; set; }
    
      [JsonProperty("name")]
      public string name { get; set; }
    
      [JsonProperty("shortName")]
      public string shortName { get; set; }
    
      [JsonProperty("description")]
      public string description { get; set; }
      #endregion
    }
    #endregion
    public enum Importance {
      LOW,
      MEDIUM,
      HIGH
    }
    
    
    #region Issue
    public class Issue {
      #region members
      [JsonProperty("id")]
      public string id { get; set; }
    
      [JsonProperty("title")]
      public string title { get; set; }
    
      [JsonProperty("description")]
      public string description { get; set; }
    
      [JsonProperty("number")]
      public int number { get; set; }
    
      [JsonProperty("points")]
      public int? points { get; set; }
    
      [JsonProperty("flagged")]
      public bool flagged { get; set; }
    
      [JsonProperty("importance")]
      public Importance importance { get; set; }
    
      [JsonProperty("type")]
      public IssueType type { get; set; }
    
      [JsonProperty("state")]
      public State state { get; set; }
    
      [JsonProperty("epic")]
      public Epic epic { get; set; }
    
      [JsonProperty("creator")]
      public User creator { get; set; }
    
      [JsonProperty("worker")]
      public User worker { get; set; }
      #endregion
    }
    #endregion
    public enum IssueType {
      STORY,
      IMPROVEMENT,
      TASK,
      BUG,
      DEPT
    }
    
    
    #region Mutation
    public class Mutation {
      #region members
      [JsonProperty("createBoard")]
      public Board createBoard { get; set; }
    
      [JsonProperty("updateBoard")]
      public Board updateBoard { get; set; }
    
      [JsonProperty("deleteBoard")]
      public string deleteBoard { get; set; }
      #endregion
    }
    #endregion
    
    #region Project
    public class Project {
      #region members
      [JsonProperty("id")]
      public string id { get; set; }
    
      [JsonProperty("name")]
      public string name { get; set; }
    
      [JsonProperty("shortName")]
      public string shortName { get; set; }
    
      [JsonProperty("backlogs")]
      public List<BackLog> backlogs { get; set; }
    
      [JsonProperty("boards")]
      public List<Board> boards { get; set; }
      #endregion
    }
    #endregion
    
    #region Query
    public class Query {
      #region members
      [JsonProperty("me")]
      public User me { get; set; }
    
      [JsonProperty("boards")]
      public List<Board> boards { get; set; }
      #endregion
    }
    #endregion
    
    #region State
    public class State {
      #region members
      [JsonProperty("id")]
      public string id { get; set; }
    
      [JsonProperty("name")]
      public string name { get; set; }
      #endregion
    }
    #endregion
    
    #region Subscription
    public class Subscription {
      #region members
      [JsonProperty("time")]
      public string time { get; set; }
      #endregion
    }
    #endregion
    
    #region User
    public class User {
      #region members
      [JsonProperty("id")]
      public string id { get; set; }
    
      [JsonProperty("userName")]
      public string userName { get; set; }
    
      [JsonProperty("firstName")]
      public string firstName { get; set; }
    
      [JsonProperty("lastName")]
      public string lastName { get; set; }
      #endregion
    }
    #endregion
  }
  
}

File: Frontend/src/model/generated.ts
export type Maybe<T> = T | null;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
};

export type BackLog = {
  __typename?: 'BackLog';
  id: Scalars['ID'];
  name: Scalars['String'];
  issues: Array<Issue>;
};

export type Board = {
  __typename?: 'Board';
  id: Scalars['ID'];
  backlog: BackLog;
  states: Array<State>;
};

export type BoardInput = {
  name: Maybe<Scalars['String']>;
};

export type Epic = {
  __typename?: 'Epic';
  id: Scalars['ID'];
  name: Scalars['String'];
  shortName: Scalars['String'];
  description: Scalars['String'];
};

export enum Importance {
  Low = 'LOW',
  Medium = 'MEDIUM',
  High = 'HIGH'
}

export type Issue = {
  __typename?: 'Issue';
  id: Scalars['ID'];
  title: Scalars['String'];
  description: Scalars['String'];
  number: Scalars['Int'];
  points: Maybe<Scalars['Int']>;
  flagged: Scalars['Boolean'];
  importance: Importance;
  type: IssueType;
  state: Maybe<State>;
  epic: Maybe<Epic>;
  creator: Maybe<User>;
  worker: Maybe<User>;
};

export enum IssueType {
  Story = 'STORY',
  Improvement = 'IMPROVEMENT',
  Task = 'TASK',
  Bug = 'BUG',
  Dept = 'DEPT'
}

export type Mutation = {
  __typename?: 'Mutation';
  createBoard: Board;
  updateBoard: Board;
  deleteBoard: Maybe<Scalars['String']>;
};


export type MutationCreateBoardArgs = {
  name: Scalars['String'];
};


export type MutationUpdateBoardArgs = {
  id: Scalars['ID'];
  board: BoardInput;
};


export type MutationDeleteBoardArgs = {
  id: Scalars['ID'];
};

export type Project = {
  __typename?: 'Project';
  id: Scalars['ID'];
  name: Scalars['String'];
  shortName: Scalars['String'];
  backlogs: Array<BackLog>;
  boards: Array<Board>;
};

export type Query = {
  __typename?: 'Query';
  me: Maybe<User>;
  boards: Maybe<Array<Board>>;
};

export type State = {
  __typename?: 'State';
  id: Scalars['ID'];
  name: Scalars['String'];
};

export type Subscription = {
  __typename?: 'Subscription';
  time: Maybe<Scalars['String']>;
};

export type User = {
  __typename?: 'User';
  id: Scalars['ID'];
  userName: Scalars['String'];
  firstName: Scalars['String'];
  lastName: Scalars['String'];
};
